docs/guide/blog/comment-blog.md

```diff
@@ -0,0 +1,548 @@
+)
+
+func (k msgServer) CreateComment(goCtx context.Context, msg *types.MsgCreateComment) (*types.MsgCreateCommentResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	BlockHeight := CreatedAt() // Invoke method to get the block height of Post
+	PostID := CreatedId() // Invoke method to get the Post ID
+
+	// Create variable of type comment
+	var comment = types.Comment{
+		Creator: msg.Creator,
+		Id:      msg.Id,
+		Body:    msg.Body,
+		Title:   msg.Title,
+		BlogID:  msg.BlogID,
+		CreatedAt: ctx.BlockHeight(),
+	}
+
+	// Check if the Post Exists for which a comment is being created
+	if comment.BlogID > PostID {
+		return nil, sdkerrors.Wrapf(types.ErrID, "Post Blog Id %d does not exist for which comment with Blog Id %d was made", PostID, comment.BlogID)
+	}
+
+	BlockHeight = BlockHeight + 100 // Hardcoded value to 100. This can be changed as per requirement.
+	
+	// Check if the comment is older than the Post. If more than 100 blocks, then return error.
+	if comment.CreatedAt > BlockHeight {
+		return nil, sdkerrors.Wrapf(types.ErrCommentOld, "Comment created at %d is older than post created at %d", comment.CreatedAt, BlockHeight)
+	} else {
+		id := k.AppendComment(ctx, comment)
+		return &types.MsgCreateCommentResponse{Id: id}, nil
+	}
+}
+```
+
+When Comment's validity is checked, it throws 2 error messages - `ErrID` and `ErrCommendOld`. You can define the error messaged by adding it to errors definition in x/blog/types/errors.go
+
+```go
+//...
+var (
+	ErrCommentOld = sdkerrors.Register(ModuleName, 1300, "")
+)
+
+var (
+	ErrID = sdkerrors.Register(ModuleName, 1400, "")
+)
+```
+
+Notice, 2 methods `CreatedAt` and `CreatedId` are being invoked inside `msg_server_create_comment.go`. They are responsible for returning the Block height and latest ID of Post respectively.
+
+Define `CreatedAt` and `CreatedId` method inside `x/blog/keeper/msg_server_create_post.go`
+
+```go
+//...
+
+var BlockHeight int64
+var BlockId uint64 
+
+func (k msgServer) CreatePost(goCtx context.Context, msg *types.MsgCreatePost) (*types.MsgCreatePostResponse, error) {
+  //...
+  var post = types.Post{
+    //...
+    CreatedAt: ctx.BlockHeight(),
+  }
+
+    // Define the globally declared `BlockHeight` variable
+    BlockHeight = post.CreatedAt
+
+    id := k.AppendPost(ctx, post)
+    //...
+
+    // After the post has been appended, fetch the latest id in globally declared `BlockId` variable
+    BlockId = id
+}
+
+// Define two functions to return BlockId and Height respectively.
+
+func CreatedAt() int64 {
+	return BlockHeight
+}
+
+func CreatedId() uint64 {
+	return BlockId
+}
+```
+
+
+## Write Data to the Store
+
+Define the `Comment` type and the `AppendComment` keeper method.
+
+When you define the `Comment` type in a proto file, Starport (with the help of `protoc`) takes care of generating the required Go files.
+
+Create the `proto/blog/comment.proto` file and define the `Comment` message:
```

Comment:
Use `starport scaffold type` instead.

Reply:
If I understand correctly the ask is to use - `starport scaffold type comment`?

I will move this part to the top when we do scaffolding for message. Makes sense?

Reply:
However `starport scaffold type comment` throws - "component type with name comment is already created (type MsgCreateComment exists)" 
Not sure if this is what you meant?

Reply:
@fadeev - I think it should be first scaffolding type, and then scaffolding message. That works - Thanks!

Reply:
> However starport scaffold type comment throws - "component type with name comment is already created (type MsgCreateComment exists)"

This is probably a bug. I'll raise an issue about this.